sdk/component/logger.go

```diff
@@ -0,0 +1,114 @@
+package component
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+)
+
+const (
+	logKeyError               = "error"
+	logKeyPanicStackTrace     = "panic_stack_trace"
+	logKeySDKVersion          = "sdk_version"
+	logKeyComponentName       = "component_name"
+	logKeyComponentType       = "component_type"
+	logKeyNumRawFindings      = "num_raw_findings"
+	logKeyRawFinding          = "raw_finding"
+	logKeyNumParsedFindings   = "num_parsed_findings"
+	logKeyNumFilteredFindings = "num_filtered_findings"
+	logKeyNumEnrichedFindings = "num_enriched_findings"
+
+	ctxLoggerKey = ctxgKey("logging")
+)
+
+type (
+	// Logger exposes an slog.Logger compatible logger contract.
+	Logger interface {
+		Debug(msg string, keyvals ...any)
+		Info(msg string, keyvals ...any)
+		Warn(msg string, keyvals ...any)
+		Error(msg string, keyvals ...any)
+		With(args ...any) Logger
+	}
+
+	ctxgKey string
+
+	defaultLogger struct {
```

Comment:
The default logger uses plain slog.Logger which is quite good.

The user can still customise this if needed, for example, using logrus or zap.